etna/datasets/tsdataset.py

```diff
@@ -295,6 +297,13 @@ def make_future(
                             f"NaN-s will be used for missing values"
                         )
 
+        # remove components and quantiles
```

Comment:
Both of them are not added to `raw_df`, not sure that we really need to drop the explicitly

Reply:
In our fixtures we create some datasets, giving components and quantiles in `raw_df`. For example, you can look at fixtures that was used in `test_make_future_removes_quantiles` and `test_make_future_removes_target_components`. We can try to fix this fixtures, I think.

For components we can in some sense guarantee that there will be no components (because they are added using special method). But for quantiles we can't make such guarantee.

Reply:
I think a good solution should consist of two steps:
- Rework quantiles to be like components
- Fix our fixtures to follow the procedure of adding quantiles and components

We can probably create some task for this.

Offtop, but I also think that we should somehow make a more formal restriction on columns that can be passed into `df` of `TSDataset.__init__`. Because we suggest that it is only endog, but we don't check this and in a lot of inner code don't follow this rule, so it can lead to confusing situations.

Reply:
Agree, the right solution is to rework quantiles. What kind of restrictions do you mean? Like allow only target?

Reply:
What are we going to do with this issue? Leave dropping this way and fix it after fixing quantiles?

Reply:
We should discuss do we want this restriction or not, but the proposal is to allow in `df` to be only columns `timestamp`, `segment`, `target`. We can also generate a good error message if smth went wrong.

Reply:
> What are we going to do with this issue? Leave dropping this way and fix it after fixing quantiles?

Probably, we can do like this, yes. This piece of code won't break anything (as I understand), but can safe us for now from some strange moments. We can add todo about fixing it after reworking quantiles. But we have a problem that we aren't really fixing todos. We can create a task on fixing quantiles and point out in that task to remove these redundant lines.

Reply:
1. Let's leave TODO
2. We can restrict df to have only timestamp, segment, target -- I guess it should prevent people adding exogs incorrectly. However it seems that in some places we create dataset out of df with extra columns ourselvs